### PR TITLE
feat/FWS-69/User_receives_notification_when_appointment_cancels

### PR DESCRIPTION
**Pull request by Eli293**
JIRA: link to jira ticket
https://champlainsaintlambert.atlassian.net/browse/FWS-69
## Context:
This ticket is about the notification functionality via Gmail using Javamail Sender. The function gets called whenever an appointment cancels.

## Changes

- Added POST functionality to new send endpoint
- This affects managing member's appointments use case

## UI Changes
NEW
image

### CODE REVIEW

File: fitwsarah_frontend/src/views/PersonalTrainerPanel/TrainerAppointments.js [hunk 1]
```diff
@@ -178,6 +178,76 @@ function AdminAccounts() {
...
+            const data = await response.json();
+            sendCancelMail(time, date, data.email);
+        } catch (error) {
+            console.error(error);
+        }
+    };
+    
+    
+    const sendCancelMail = (time, date, email) => {
+        const cancelMail = {subject: "Cancelled Appointment", message:`Hello, your appointment at ${time} for ${date} has been cancelled.`};
```
Comment (AlexeiTimbro) [inline fitwsarah_frontend/src/views/PersonalTrainerPanel/TrainerAppointments.js:226]: Can you elaborate on what the subject does in this situation for the const cancelMail?

**Review by AlexeiTimbro — APPROVED** [review]
Good job Elias i left a comment under here

File: fitwsarah_frontend/src/views/PersonalTrainerPanel/TrainerAppointments.js (hunk 1 above)
Comment (Eli293) [reply fitwsarah_frontend/src/views/PersonalTrainerPanel/TrainerAppointments.js:226]: The subject is the header of the email sent.

File: src/main/java/com/fitwsarah/fitwsarah/authenticationsubdomain/presentationlayer/MailController.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+@RestController
+@RequestMapping("api/v1/send")
+public class MailController {
+
+
+    @Autowired
+    private MailService mailService;
+
+    @PostMapping("/{email}")
```
Comment (Abdur-Siddiqui) [inline src/main/java/com/fitwsarah/fitwsarah/authenticationsubdomain/presentationlayer/MailController.java:17]: could you pls explain the code ere.

**Review by Abdur-Siddiqui — APPROVED** [review]
Great work

File: src/main/java/com/fitwsarah/fitwsarah/authenticationsubdomain/presentationlayer/MailController.java (hunk 2 above)
Comment (Eli293) [reply src/main/java/com/fitwsarah/fitwsarah/authenticationsubdomain/presentationlayer/MailController.java:17]: This is simply an endpoint to create post requests for sending emails. It basically takes an email as a path variable and sends the message towards it.

**Pull request merged**